backend/src/auth/auth.ts

```diff
@@ -29,6 +34,52 @@ passport.use(
     )
 );
 
+// passport.use(
+//     'signupCounselor',
+//     new customStrategy(
+//       {
+//         usernameField: 'email',
+//         passwordField: 'password',
+//         urlField: 'url',
+//       },
+//       async (email: string, password: string, url: string, done: any) => {
+//         try {
+//           const user = await UserModel.create({ email, password });
+//           const counselor = await Counselor.create({ user: user._id });
+//           user.counselor = counselor._id;
+//           await user.save();
+
+//           return done(null, user);
+//         } catch (error) {
+//           done(error);
+//         }
+//       }
+//     )
+// );
+
+passport.use(
```

Comment:
@rishabluthra we basically created an entire new passport custom strategy for counselor and we would have to for each creation of a new user (with a user type). Is there a better way to do this?